### ISSUE TITLE
CS8600: Should report on methods that returns ConfiguredTaskAwaitable

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn#39417]
**Version Used**: Visual Studio 2019 Version 16.3.5

**Steps to Reproduce**:

```csharp
static class TaskExtensions
{
    public static ConfiguredTaskAwaitable<TResult> NoSync<TResult>(this Task<TResult> task)
    {
        return task.ConfigureAwait(false);
    }
}

class Request
{
    public string? Name { get; }
    public Task<string?> GetName() => Task.FromResult(Name);
}

class Handler
{
    public async Task Handle(Request request)
    {
        string a = request.Name;    // <-- CS8600
        string b = await request.GetName().ConfigureAwait(false);    // <-- CS8600
        string c = await request.GetName().NoSync();    // <-- No warning here
    }
}
```

**Expected Behavior**:
`CS8600` to be reported when using `NoSync` method extension.

**Actual Behavior**:
`CS8600` is not reported when using `NoSync` method extension.

**Comment by skarllot** [comment]
Can I include a unit test for this case?

**Comment by jcouv** [comment]
@skarllot Sorry for late reply. Sure, feel free to add a unittest. It could go in `NullableReferenceTypeTest.cs`

**Comment by skarllot** [comment]
I wrote the tests and appears that the analyzer on master already fixed that issue.

**Comment by RikkiGibson** [comment]
Probably the same root cause as #40452, where `await`ing a call to a nullable-reinterred generic method such as `NoSync` would cause us to analyze using the originally inferred type arguments (i.e. type arguments that are unaware of nullability).
